cscs-checks/analytics/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,138 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+@rfm.simple_test
+class HPCGCheckRef(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+
+        self.descr = 'HPCG reference benchmark'
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['craype-hugepages8M']
+        self.build_system = 'Make'
+        self.build_system.options = ['arch=MPI_GCC_OMP']
+        self.sourcesdir = 'https://github.com/hpcg-benchmark/hpcg.git'
+
+        self.executable = 'bin/xhpcg'
+        self.executable_opts = ['--nx=104', '--ny=104', '--nz=104', '-t2']
+        output_file = sn.getitem(sn.glob('HPCG*.txt'), 0)
+        self.sanity_patterns = sn.assert_eq(4, sn.count(
+            sn.findall(r'PASSED', output_file)))
+
+        self.num_cpus_per_task = 1
+        self.system_num_tasks = {
+            'daint:mc':  36,
+            'daint:gpu': 12,
+            'dom:mc':  36,
+            'dom:gpu': 12,
+        }
+
+        self.reference = {
+            'daint:gpu': {
+                'perf': (7.6, -0.1, None)
```

Comment:
Don't use `perf`, but rather a more descriptive name for the performance variable. `perf` was just a generic placeholder I've put it in some very early examples, and then survived and multiplied thanks to copy-paste...

Reply:
I suggest adding another element in the tuple that denotes the units. This is feasible from the current version only, so you should put `@rfm.required_version('>=2.16-dev0') on the top of the test.